From f35068c7c8d376c45e7946d210afe1ca1d8cb12d Mon Sep 17 00:00:00 2001
From: Alejandro Mery <amery@geeks.cl>
Date: Fri, 13 Jul 2012 14:02:18 +0200
Subject: [PATCH 12/27] drm: mali: fix for 3.4

(cherry picked from commit 5bba4fb83f9f68d2e84ec7b83ce334f7fd5dd704)
---
 drivers/gpu/drm/mali/mali_drv.c | 26 ++++++++++++++------------
 1 file changed, 14 insertions(+), 12 deletions(-)

diff --git a/drivers/gpu/drm/mali/mali_drv.c b/drivers/gpu/drm/mali/mali_drv.c
index e88873e..af8c014 100644
--- a/drivers/gpu/drm/mali/mali_drv.c
+++ b/drivers/gpu/drm/mali/mali_drv.c
@@ -13,12 +13,13 @@
  * Implementation of the Linux device driver entrypoints for Mali DRM
  */
 
+#include <linux/module.h>
 #include <linux/vermagic.h>
 #include "drmP.h"
 #include "mali_drv.h"
 
 
-void mali_drm_preclose(struct drm_device *dev)
+void mali_drm_preclose(struct drm_device *dev, struct drm_file *file_priv)
 {
 }
 
@@ -26,7 +27,7 @@ void mali_drm_lastclose(struct drm_device *dev)
 {
 }
 
-static int mali_drm_suspend(struct drm_device *dev)
+static int mali_drm_suspend(struct drm_device *dev, pm_message_t state)
 {
 	return 0;
 }
@@ -46,6 +47,16 @@ static int mali_drm_unload(struct drm_device *dev)
 	return 0;
 }
 
+static struct file_operations mali_fops = {
+	 .owner = THIS_MODULE,
+	 .open = drm_open,
+	 .release = drm_release,
+	 .unlocked_ioctl = drm_ioctl,
+	 .mmap = drm_mmap,
+	 .poll = drm_poll,
+	 .fasync = drm_fasync,
+};
+
 static struct drm_driver driver = 
 {
 	.driver_features = DRIVER_BUS_PLATFORM,
@@ -58,17 +69,8 @@ static struct drm_driver driver =
 	.lastclose = mali_drm_lastclose,
 	.suspend = mali_drm_suspend,
 	.resume = mali_drm_resume,
-
 	.ioctls = NULL,
-	.fops = {
-		 .owner = THIS_MODULE,
-		 .open = drm_open,
-		 .release = drm_release,
-		 .unlocked_ioctl = drm_ioctl,
-		 .mmap = drm_mmap,
-		 .poll = drm_poll,
-		 .fasync = drm_fasync,
-	},
+	.fops = &mali_fops,
 	.name = DRIVER_NAME,
 	.desc = DRIVER_DESC,
 	.date = DRIVER_DATE,
-- 
1.9.1

